mono/metadata/object.c

```diff
@@ -7195,21 +7195,19 @@ mono_string_to_utf16 (MonoString *s)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
-	char *as;
-
 	if (s == NULL)
 		return NULL;
 
-	as = (char *)g_malloc ((s->length * 2) + 2);
-	as [(s->length * 2)] = '\0';
-	as [(s->length * 2) + 1] = '\0';
+	int const length = s->length;
 
-	if (!s->length) {
-		return (gunichar2 *)(as);
-	}
+	gunichar2 * const as = (gunichar2 *)g_malloc ((length * 2) + 2);
+	as [length] = 0;
+
+	if (!length)
+		return as;
```

Comment:
You could reduce the number of line by having
```
if (length)
    memcpy (as, mono_string_chars(s), length * sizeof(gunichar2));
as[length] = 0;
return as;
```

Reply:
I considered it. ok.